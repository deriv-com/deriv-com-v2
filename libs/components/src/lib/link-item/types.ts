import { type VariantProps } from 'class-variance-authority';
import { HTMLAttributes } from 'react';
import { linkItemCva } from './link-item.classnames';
import { QuillSvgProps } from '@deriv/quill-icons';
import { ExcludeNull } from '@deriv/quill-design';

export type QuillIconComponent = React.ForwardRefExoticComponent<
  Omit<QuillSvgProps, 'ref'>
>;
export interface LinkItemProps
  extends HTMLAttributes<HTMLSpanElement>,
    ExcludeNull<
      VariantProps<typeof linkItemCva>,
      'colorStyle' | 'size' | 'disabled'
    > {
  icon?: QuillIconComponent;
  hasIcon?: boolean;
}
export type LinkItemSize = 'caption' | 'sm' | 'md' | 'lg' | 'xl';

export type colorStyle = LinkItemProps['colorStyle'];

export type LinkItemTestSize = LinkItemProps['size'];

export type LinkItemDisabled = LinkItemProps['disabled'];
