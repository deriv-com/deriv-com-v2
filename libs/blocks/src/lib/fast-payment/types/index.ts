import { ComponentProps, ReactNode } from 'react';
import { CardsContainerProps } from '@deriv-com/components';
import { Section } from '@deriv/quill-design';

//todo: removed [LinkProps] from utils/generics and added here. we need to update this because we are using this on multiple places
interface LinkProps {
  content: ReactNode;
  href: string;
}

type SectionProps = Omit<ComponentProps<typeof Section>, 'content' | 'title'>;

export type FastPaymentProps = SectionProps & {
  title: ReactNode;
  description?: ReactNode;
  link?: LinkProps;
  disclaimer?: ReactNode;
  content: Omit<CardsContainerProps<'ContentBottom'>, 'variant'>;
  className?: string;
};
