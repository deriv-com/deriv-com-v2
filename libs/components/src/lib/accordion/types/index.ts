import { ReactNode } from 'react';

export interface AccordionProps {
  id?: string;
  className?: string;
  title: string;
  subtitle?: string;
  icon?: ReactNode;
  content?: () => ReactNode;
  expanded?: boolean;
  customContent?: () => ReactNode;
  divider?: AccordionDivider;
  contentClass?: string;
  background?: BgVariant;
  onExpand?: (e: boolean, i: string) => void;
}
export type BgVariant = 'light' | 'gray';
export type AccordionDivider = 'none' | 'both' | 'bottom';
