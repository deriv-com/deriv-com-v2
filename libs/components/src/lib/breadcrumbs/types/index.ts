import { LinkProps } from '@deriv-com/utils';

export type BreadcrumbsProps = {
  size?: 'md' | 'sm';
  links?: LinkProps[];
  className?: string;
};
