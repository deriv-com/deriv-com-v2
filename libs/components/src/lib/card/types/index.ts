import { ReactNode } from 'react';

export type StatCardVariant = 'white' | 'dark' | 'coral' | 'transparent';
export type CardColorVariant = 'light' | 'gray';
export type CardAlignVariant = 'start' | 'center';
export type CardSizeVariant = 'sm' | 'md' | 'lg';

interface BaseCardContent {
  header?: string;
}
//todo: removed [LinkProps] from utils/generics and added here. we need to update this because we are using this on multiple places
interface LinkProps {
  content: ReactNode;
  href: string;
}

export interface StatCardContent extends BaseCardContent {
  color?: StatCardVariant;
  description: string;
}

export interface CardContent extends BaseCardContent {
  icon?: ReactNode;
  description?: string;
  children?: ReactNode;
  link?: LinkProps;
  content?: ReactNode;
  color: CardColorVariant;
  align: CardAlignVariant;
  size: CardSizeVariant;
  className?: string;
  contentClassName?: string;
  nonContentClassName?: string;
}

export type MarketStatus = 'up' | 'down' | 'remain' | 'closed';
export interface LiveMarketContent {
  instrumentIcon: ReactNode;
  instrument: string;
  changePercentage: string;
  status: MarketStatus;
  bidPrice: string;
  askPrice: string;
  spread: string;
}
