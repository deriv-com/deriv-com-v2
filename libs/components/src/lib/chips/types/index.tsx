import { ReactNode } from 'react';

export type ChipsColorVariant = 'coral' | 'dark' | 'black' | 'transparent';
export type ChipsSizeVariant = 'sm' | 'md' | 'lg';
export type ChipsTextVariant = 'sm' | 'md' | 'lg';
export type ChipsTextColorVariant = 'black' | 'white' | 'transparent';
export type ChipsVariants =
  | 'default'
  | 'hover'
  | 'active'
  | 'selected'
  | 'disabled';
export interface ChipsProps {
  label?: string;
  icon?: ReactNode;
  children?: ReactNode;
  className?: string;
  color?: string;
  size?: string;
  textColor?: string;
  state?: ChipsVariants;
  disabled?: boolean;
}
