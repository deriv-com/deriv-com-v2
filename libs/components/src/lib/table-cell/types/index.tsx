import { ReactNode } from 'react';

export type TableCellVariant = 'header' | 'symbol' | 'text';
export type TableCellColorVariant = 'white' | 'coral';
export type TableWidthVariant = 'xl' | 'lg' | 'md' | 'sm';

export interface TableCellContent {
  header?: string;
  label?: string | ReactNode;
  children?: ReactNode;
  icon?: ReactNode;
  description?: string;
  color?: TableCellColorVariant;
  className?: string;
  width?: TableWidthVariant;
}
