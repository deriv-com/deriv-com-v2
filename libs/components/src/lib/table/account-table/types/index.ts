import { ReactNode } from 'react';

interface SimpleListItem {
  type: 'simple';
  text: string;
}

interface CompoundListItem {
  type: 'compound';
  text: string;
  icon: ReactNode;
  listIcon: ReactNode;
  items: string[];
}

export interface ColContent {
  subheader: string;
  listItems: Array<SimpleListItem | CompoundListItem>;
}

interface RowContent {
  cols: ColContent[];
}

export interface TableContent {
  header: string;
  description: string;
  cta: ReactNode;
  rows: RowContent[];
}
