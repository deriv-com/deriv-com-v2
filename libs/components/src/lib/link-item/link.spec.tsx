import { render, screen } from '@testing-library/react';
import LinkItem from '.';
import { colorStyle, LinkItemDisabled, LinkItemTestSize } from './types';
import { StandaloneArrowLeftBoldIcon } from '@deriv/quill-icons/Standalone';
import '@testing-library/jest-dom';

describe('LinkItem', () => {
  it('should render correctly', () => {
    render(<LinkItem>Click me</LinkItem>);
    const LinkElement = screen.getByText(/click me/i);
    expect(LinkElement).toBeInTheDocument();
  });

  it('should render the classname', () => {
    render(<LinkItem className="px-300">Click me</LinkItem>);
    const LinkElement = screen.getByText(/click me/i);
    expect(LinkElement).toHaveClass('px-300');
  });

  it('should render chevron icon when hasIcon prop is true', () => {
    render(<LinkItem hasIcon>Click me</LinkItem>);
    const icon = screen.getByTestId('dt-link-item-chevron');
    expect(icon).toBeInTheDocument();
  });

  it('should render icon when a icon is pass', () => {
    render(<LinkItem icon={StandaloneArrowLeftBoldIcon}>Click me</LinkItem>);
    const icon = screen.getByTestId('dt-link-item-icon');
    expect(icon).toBeInTheDocument();
  });

  const colorStyles: colorStyle[] = ['black', 'white'];
  const sizes: LinkItemTestSize[] = ['caption', 'sm', 'md', 'lg', 'xl'];

  colorStyles.forEach((colorStyle) => {
    sizes.forEach((size) => {
      it(`should render correctly with ${size} size and color ${colorStyle}`, () => {
        const { container } = render(
          <LinkItem size={size} colorStyle={colorStyle}>
            Click me - {size}
          </LinkItem>,
        );
        expect(container).toMatchSnapshot();
      });
    });
  });

  const disabled: LinkItemDisabled[] = [true, false];
  disabled.forEach((dis) => {
    it(`should render correctly with disabled ${dis}`, () => {
      render(<LinkItem disabled={dis}>Click me disabled</LinkItem>);
      const LinkElement = screen.getByText('Click me disabled');
      expect(LinkElement).toMatchSnapshot();
    });
  });
});
