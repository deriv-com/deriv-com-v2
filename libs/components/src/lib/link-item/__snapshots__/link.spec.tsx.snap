// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkItem > should render correctly with caption size and color black 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-400 text-50"
  >
    Click me - 
    caption
  </span>
</div>
`;

exports[`LinkItem > should render correctly with caption size and color white 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-50 hover:underline hover:cursor-pointer active:underline p-400 text-50"
  >
    Click me - 
    caption
  </span>
</div>
`;

exports[`LinkItem > should render correctly with disabled false 1`] = `
<span
  class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-400 text-50"
>
  Click me disabled
</span>
`;

exports[`LinkItem > should render correctly with disabled true 1`] = `
<span
  class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-400 text-50 opacity-300 cursor-none pointer-events-none"
>
  Click me disabled
</span>
`;

exports[`LinkItem > should render correctly with lg size and color black 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-800 text-100"
  >
    Click me - 
    lg
  </span>
</div>
`;

exports[`LinkItem > should render correctly with lg size and color white 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-50 hover:underline hover:cursor-pointer active:underline p-800 text-100"
  >
    Click me - 
    lg
  </span>
</div>
`;

exports[`LinkItem > should render correctly with md size and color black 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-600 text-75"
  >
    Click me - 
    md
  </span>
</div>
`;

exports[`LinkItem > should render correctly with md size and color white 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-50 hover:underline hover:cursor-pointer active:underline p-600 text-75"
  >
    Click me - 
    md
  </span>
</div>
`;

exports[`LinkItem > should render correctly with sm size and color black 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-400 text-50"
  >
    Click me - 
    sm
  </span>
</div>
`;

exports[`LinkItem > should render correctly with sm size and color white 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-50 hover:underline hover:cursor-pointer active:underline p-400 text-50"
  >
    Click me - 
    sm
  </span>
</div>
`;

exports[`LinkItem > should render correctly with xl size and color black 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-1400 hover:underline hover:cursor-pointer active:underline p-1200 text-300"
  >
    Click me - 
    xl
  </span>
</div>
`;

exports[`LinkItem > should render correctly with xl size and color white 1`] = `
<div>
  <span
    class="inline-flex items-center gap-500 text-solid-slate-50 hover:underline hover:cursor-pointer active:underline p-1200 text-300"
  >
    Click me - 
    xl
  </span>
</div>
`;
